### PR TITLE
Add PATCH to Azure Firewall supported operations

### PR DESCRIPTION
**Pull request by fanymanea**
We are adding PATCH to the list of supported operations for Azure Firewall.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by number213** [comment]
Please ignore error in `NetworkValidation`, it's not related to this PR, it failed due to NuGet not being able to resolve dependencies. I've already reported this error to SDK & Azure PowerShell teams.

```
D:\a\1\s\Out\azure-powershell\src\Network\Network\Network.csproj : error NU1605: Detected package downgrade: Microsoft.Rest.ClientRuntime from 2.3.20 to 2.3.19. Reference the package directly from the project to select a different version.  [D:\a\1\s\Out\azure-powershell\src\Network\Network.sln]
```

**Comment by fanymanea** [comment]
> SDK & Azure PowerShell team

Then can you please help me complete this PR?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/azureFirewall.json [hunk 1]
```diff
@@ -192,6 +192,58 @@
...
+              "$ref": "#/definitions/AzureFirewall"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Update Azure Firewall Tags": {
+            "$ref": "./examples/AzureFirewallUpdateTags.json"
+          }
+        },
+        "x-ms-long-running-operation": true
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/azureFirewall.json]: Per specs, operation should be marked as long-running if it's returning codes 201 or 202.
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation
I've tried various PATCHes, including Azure Firewall, all of them returned 200 and updated resources synchronously, so I don't think this option is needed here

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/azureFirewall.json (hunk 1 above)
Comment (fanymanea) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/azureFirewall.json]: Thank you! You're right. I updated the spec.

**Review by anton-evseev — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**